datafusion/common/src/scalar.rs

```diff
@@ -1032,8 +1020,11 @@ impl ScalarValue {
             ScalarValue::Binary(v) => v.is_none(),
             ScalarValue::FixedSizeBinary(_, v) => v.is_none(),
             ScalarValue::LargeBinary(v) => v.is_none(),
-            ScalarValue::Fixedsizelist(v, ..) => v.is_none(),
-            ScalarValue::List(arr) => arr.len() == arr.null_count(),
+            // arr.len() should be 1 for a list scalar, but we don't seem to
```

Comment:
why `arr.len()` should be 1? I thought it can be any kind of `ListArray`

Reply:
If `arr.len() > 1`, it isn't a scalar. `arr` is a `ListArray` or `FixedSizeListArray`, so `arr.len() == 1` means there is one list.